src/main/java/net/discordjug/javabot/util/WebhookUtil.java

```diff
@@ -137,6 +137,59 @@ private static CompletableFuture<ReadonlyMessage> mirrorMessageToWebhook(@NotNul
 				});
 	}
 
+	/**
+	 * Resends a specific message using a webhook with a custom content.
+	 *
+	 * @param webhook           the webhook used for sending the message
+	 * @param originalMessage   the message to copy
+	 * @param newMessageContent the new (custom) content
+	 * @param threadId          the thread to send the message in or {@code 0} if the
+	 *                          message should be sent directly
+	 * @param components        A nullable list of {@link LayoutComponent}s.
+	 * @param attachments            A list of {@link Message.Attachment}s.
+	 * @return a {@link CompletableFuture} representing the action of sending
+	 * the message
+	 */
+	public static CompletableFuture<ReadonlyMessage> mirrorMessageToWebhookWithAttachments(@NotNull Webhook webhook, @NotNull Message originalMessage, String newMessageContent, long threadId, @Nullable List<LayoutComponent> components, List<Message.Attachment> attachments) {
```

Comment:
Please don't duplicate code. Instead, you can add an `attachments` parameter to the existing method and create a new overload with the old signature.